support-frontend/assets/helpers/abTests/__tests__/abtestTest.ts

```diff
@@ -21,840 +18,303 @@ jest.mock('ophan', () => ({
 }));
 
 // ----- Tests ----- //
-const emptySettings = {
-	switches: {
-		experiments: {},
-	},
-	amounts: {
-		GBPCountries: {},
-		UnitedStates: {},
-	},
-};
-
-const acquisitionDataMockTestControl = {
-	componentId: 'acquisitions-subscription-banner',
-	componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
-	abTest: {
-		name: 'mockTest',
-		variant: 'control',
-	},
-};
-
-const acquisitionDataMockTestVariant = {
-	componentId: 'acquisitions-subscription-banner',
-	componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
-	abTest: {
-		name: 'mockTest',
-		variant: 'variant',
-	},
-};
-
-const acquisitionDataMockTest2Control = {
-	componentId: 'acquisitions-subscription-banner',
-	componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
-	abTest: {
-		name: 'mockTest2',
-		variant: 'control',
-	},
-};
-
-const acquisitionDataMockTest2Variant = {
-	componentId: 'acquisitions-subscription-banner',
-	componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
-	abTest: {
-		name: 'mockTest2',
-		variant: 'variant',
-	},
-};
-
-const mockTestControl = `/test.html?acquisitionData=${encodeURI(
-	JSON.stringify(acquisitionDataMockTestControl),
-)}`;
-
-const mockTestVariant = `/test.html?acquisitionData=${encodeURI(
-	JSON.stringify(acquisitionDataMockTestVariant),
-)}`;
-
-const mockTest2Control = `/test.html?acquisitionData=${encodeURI(
-	JSON.stringify(acquisitionDataMockTest2Control),
-)}`;
-
-const mockTest2Variant = `/test.html?acquisitionData=${encodeURI(
-	JSON.stringify(acquisitionDataMockTest2Variant),
-)}`;
-
-describe('basic behaviour of init', () => {
+
+describe('init', () => {
 	Object.defineProperty(window, 'matchMedia', {
-		value: jest.fn().mockImplementation(() => ({
+		value: jest.fn().mockReturnValue({
 			matches: false,
-		})),
+		}),
 	});
+
+	// Common arguments to init
+	const mvt = 123456;
+	const country = 'GB';
+	const countryGroupId = GBPCountries;
+	const emptySettings = {
+		switches: {
+			experiments: {},
+		},
+		amounts: {
+			GBPCountries: {},
+			UnitedStates: {},
+		},
+	} as Settings;
+
 	afterEach(() => {
 		window.localStorage.clear();
 	});
-	it('The referrerControlled user should be allocated to the control bucket', () => {
-		document.cookie = 'GU_mvt_id=12346';
-		window.history.pushState({}, 'Test Title', mockTestControl);
+
+	it('assigns a user to a variant', () => {
 		const tests = {
-			mockTest: {
-				type: 'OTHER',
-				variants: [
-					{
-						id: 'control',
-					},
-					{
-						id: 'variant',
-					},
-				],
-				audiences: {
-					GB: {
-						offset: 0,
-						size: 1,
-					},
-				},
-				isActive: true,
-				referrerControlled: true,
-				seed: 0,
-			},
+			t: buildTest({ variants: [buildVariant({ id: 'control' })] }),
```

Comment:
I've factored out a number of `build` functions. This has two benefits. First, it cuts down on the number of lines of code. Secondly (and more importantly), it allows us to just specify the salient parts of the test that we want. Here, the only important part is that we have a test called `t` with a single variant `control`. This is important because we're going to explicitly assert that we get assigned to this test and variant.